homeassistant/components/http/__init__.py

```diff
@@ -195,13 +199,28 @@ def __init__(self, hass, development, api_password, ssl_certificate,
         self.app[KEY_LOGIN_THRESHOLD] = login_threshold
         self.app[KEY_DEVELOPMENT] = development
 
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.bind((server_host, server_port))
+            if server_port == 0:
```

Comment:
We should not add support for random ports to our HTTP component to make testing easier. Instead we should use the [unused port function from aiohttp test utils](http://aiohttp.readthedocs.io/en/stable/testing.html#aiohttp.test_utils.unused_port).